official/keras_application_models/model_callbacks.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Callbacks for Keras built-in application models."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import time
+
+import tensorflow as tf  # pylint: disable=g-bad-import-order
+
+from official.utils.logs import logger
+
+# Metrics to log after each batch and epoch
+_PER_BATCH_METRICS = {
+    "loss": "train_loss",
+    "acc": "train_accuracy",
+}
+_PER_EPOCH_METRICS = {
+    "loss": "train_loss",
+    "acc": "train_accuracy",
+    "val_loss": "loss",
+    "val_acc": "accuracy"
+}
+
+
+class ExamplesPerSecondCallback(tf.keras.callbacks.Callback):
+  """ExamplesPerSecond callback.
+
+  This callback records the average_examples_per_sec and
+  current_examples_per_sec during training.
+  """
+
+  def __init__(self, batch_size, every_n_steps=1, metric_logger=None):
+    self._batch_size = batch_size
+    self._every_n_steps = every_n_steps
+    self._logger = metric_logger or logger.BaseBenchmarkLogger()
+    self._global_step = 0
```

Comment:
I think this should be set in on_train_begin, in case user reuse the callback instance for several training cycle.

Reply:
Yeah, here is something actually I am not quite sure. In the last commit, I put global_step into the on_train_begin, but Karmel suggests to move it to the constructor. I thought as callback is invoked during the training phase, maybe it makes no difference here? What do you mean by "reusing the callback instance in several training cycle"?  @qlzh727 @karmel

Reply:
on_train_begin gets called each time someone starts a train loop? Which would be for each epoch for example? Shouldn't global step be maintained for the life of the callback/model, rather than reset to 0 with each train loop? Or does on_train_begin only get called once?

Reply:
I would expect on_train_begin only gets called once when a training starts, eg a fit() call. For each epoch, it will use on_epoch_begin/end method.

Reply:
From the code, it looks like on_train_begin gets called in the fit_loop, so it would be reset with each call to fit(). Since a model can be incrementally trained with fit(), we want the global step to persist across those training sessions. So, let's keep it in init for now.